devito/finite_differences/coefficients.py

```diff
@@ -0,0 +1,256 @@
+import sympy
+import numpy as np
+from cached_property import cached_property
+
+from devito.finite_differences import generate_indices
+from devito.tools import filter_ordered
+
+__all__ = ['Coefficient', 'CoefficientRules', 'default_rules']
+
+
+class Coefficient(object):
+    """
+    Prepare custom FD coefficients to pass to CoefficientRules object.
+
+    Parameters
+    ----------
+    deriv_order : integer
+        Represents the order of the derivative being taken.
```

Comment:
nitpicking: for homogeneity, you can drop all "Represents" (same in the lines below)